### ISSUE TITLE
grunt deploy thinks all the entries got updated

### ISSUE THREAD

**Issue by mzgol** [jquery/api.jquery.com#767]
No matter how many times I invoke `grunt deploy`, I still see what's at the end of this post. This doesn't happen in the jquery.com repisitory where only edited things get deployed.
```
$ grunt deploy
Running "check-modules" task

Running "xmllint:all" (xmllint) task
Lint free files: 315

Running "clean-dist" task

Running "build-posts:page" (build-posts) task
Built 2 pages.

Running "build-resources:all" (build-resources) task
Built 64 resources.

Running "build-xml-entries:all" (build-xml-entries) task
Built 308 entries.

Running "build-xml-categories" task

Running "build-xml-full" task

Running "wordpress-validate" task
Validated 94 terms.
Validated 310 posts.

Running "wordpress-sync" task
Edited category ajax.
Edited category global-ajax-event-handlers.
Edited category helper-functions.
Edited category low-level-interface.
Edited category shorthand-methods.
Edited category attributes.
... (a lot of things here)
Edited category all.
```

**Comment by arthurvr** [comment]
I noticed this too. Maybe @scottgonzalez has an idea why this is happening? :wink:

**Comment by scottgonzalez** [comment]
TLDR: All categories are always updated.

-----

WordPress doesn't provide any way for us to track the actual state of anything compared to the raw content we have on the local disk. In order to deal with this, we store a hash of the content for each post as a custom field on the post. This provides us a checksum to compare against on every deploy. In order to reduce the network overhead, gilded-wordpress provides a method that returns a list of all posts and their checksums (see [gw_get_post_paths](https://github.com/scottgonzalez/gilded-wordpress/blob/a723e66dbdbdf1ec4e65038b8cea323dca3aa5dd/gilded-wordpress.php#L34-L50)). This is how we know which posts need to be added, edited, and deleted.

Categories (actually all taxonomies), on the other hand, don't support any custom metadata like custom fields, so we can't employ a similar tactic in order to know which categories need to be added, edited, and deleted. Instead, we just get the full list of all taxonomies and walk through them. My recollection is that the content comes back in a different form than what we send, so it's not a simple comparison. As a result, we just always edit every category to ensure it has the values we want. See https://github.com/scottgonzalez/gilded-wordpress/blob/a723e66dbdbdf1ec4e65038b8cea323dca3aa5dd/lib/taxonomies.js for the full logic.

While it may be possible to optimize this in gilded-wordpress (it's been a long time since I looked at this part of the deployment, but I remember spending time trying to avoid this problem), there's nothing to do in this repo.
